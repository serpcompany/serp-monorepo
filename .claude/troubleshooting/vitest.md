https://vitest.dev/guide/common-errors.html