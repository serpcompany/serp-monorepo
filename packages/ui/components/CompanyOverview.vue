<script setup lang="ts">
import type { Company } from '@serp/types/types'

defineProps<{
  company: Company
}>()
</script>

<template>
  <div>
    <div
      class="flex flex-col justify-between gap-6 md:flex-row md:gap-3 lg:gap-4 xl:gap-6"
    >
      <!-- Right Column: Overview and Details -->
      <section class="w-full md:flex-1">
        <p class="text-md mb-4 leading-relaxed">
          {{ company?.excerpt }}
        </p>
      </section>
    </div>
  </div>
</template>
