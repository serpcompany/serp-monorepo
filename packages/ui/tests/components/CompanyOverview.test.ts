import { describe, expect, it } from 'vitest';
import CompanyOverview from '../../components/CompanyOverview.vue';
import ComponentRender from '../componentRender';

describe('CompanyOverview Snapshot', () => {
  it.each([
    [
      'with a normal company excerpt',
      {
        props: {
          company: {
            excerpt: 'Leading the industry with innovative solutions.'
          }
        }
      }
    ],
    ['with an empty excerpt', { props: { company: { excerpt: '' } } }],
    [
      'with a long company excerpt',
      {
        props: {
          company: {
            excerpt: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. '
              .repeat(5)
              .trim()
          }
        }
      }
    ]
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `CompanyOverview ${desc}`,
        options,
        CompanyOverview
      );
      expect(html).toMatchSnapshot();
    }
  );
});
