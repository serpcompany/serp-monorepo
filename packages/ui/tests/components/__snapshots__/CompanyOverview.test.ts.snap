// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyOverview Snapshot > renders with a long company excerpt correctly 1`] = `
<div>
  <div class="
    flex
    flex-col
    gap-6
    justify-between
    lg:gap-4
    md:flex-row
    md:gap-3
    xl:gap-6
  ">
    <!-- Right Column: Overview and Details -->
    <section class="
      md:flex-1
      w-full
    ">
      <p class="
        leading-relaxed
        mb-4
        text-md
      ">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
    </section>
  </div>
</div>
`;

exports[`CompanyOverview Snapshot > renders with a normal company excerpt correctly 1`] = `
<div>
  <div class="
    flex
    flex-col
    gap-6
    justify-between
    lg:gap-4
    md:flex-row
    md:gap-3
    xl:gap-6
  ">
    <!-- Right Column: Overview and Details -->
    <section class="
      md:flex-1
      w-full
    ">
      <p class="
        leading-relaxed
        mb-4
        text-md
      ">
        Leading the industry with innovative solutions.
      </p>
    </section>
  </div>
</div>
`;

exports[`CompanyOverview Snapshot > renders with an empty excerpt correctly 1`] = `
<div>
  <div class="
    flex
    flex-col
    gap-6
    justify-between
    lg:gap-4
    md:flex-row
    md:gap-3
    xl:gap-6
  ">
    <!-- Right Column: Overview and Details -->
    <section class="
      md:flex-1
      w-full
    ">
      <p class="
        leading-relaxed
        mb-4
        text-md
      "></p>
    </section>
  </div>
</div>
`;
